mbed_targets/mbed_target.py

```diff
@@ -0,0 +1,95 @@
+class MbedTarget:
+    """Representation of an Mbed Target.
+
+    Attributes:
+        board_type: Type of board the target represents.
+        board_name: Human readable name.
+        product_code: Uniquely identifies a platform for the online compiler.
+        target_type: Identifies if the target is a module or a platform.
+        slug: Identifies a platform's page on the mbed website, used with target_type to identify the target.
+        build_variant: Build variant for the compiler.
+        mbed_os_support: Mbed OS versions supported.
+        mbed_enabled: Enabled Mbed OS support.
+    """
+
+    board_type: str
+    board_name: str
+    product_code: str
+    target_type: str
+    slug: str
+    build_variant: Tuple[str, ...]
+    mbed_os_support: Tuple[str, ...]
+    mbed_enabled: Tuple[str, ...]
+
+    @classmethod
+    def from_online_target_entry(cls, target_entry: dict) -> "MbedTarget":
+        """Create a new instance of MbedTarget from an online target database entry.
+
+        Args:
+            target_entry: A single entity retrieved from the target API.
+        """
+        target_attrs = target_entry.get("attributes", {})
+        target_features = target_attrs.get("features", {})
+
+        return cls(
+            board_type=target_attrs.get("board_type", ""),
+            board_name=target_attrs.get("name", ""),
+            mbed_os_support=tuple(target_features.get("mbed_os_support", [])),
+            mbed_enabled=tuple(target_features.get("mbed_enabled", [])),
+            product_code=target_attrs.get("product_code", ""),
+            target_type=target_attrs.get("target_type", ""),
+            slug=target_attrs.get("slug", ""),
+            # NOTE: Currently we hard code the build variant for a single board type.
+            # This is simply so we can demo the tools to PE. This must be removed and replaced with a proper mechanism
+            # for determining the build variant for all platforms. We probably need to add this information to the
+            # targets database.
+            build_variant=cast(Tuple, ("S", "NS") if "lpc55s69" in target_attrs.get("board_type", "") else ()),
+        )
+
+    @classmethod
+    def from_offline_target_entry(cls, target_entry: dict) -> "MbedTarget":
+        """Construct an MbedTarget with data from the offline database snapshot."""
+        return cls(
+            board_type=target_entry.get("board_type", ""),
+            board_name=target_entry.get("board_name", ""),
+            product_code=target_entry.get("product_code", ""),
+            target_type=target_entry.get("target_type", ""),
+            slug=target_entry.get("slug", ""),
+            mbed_os_support=tuple(target_entry.get("mbed_os_support", [])),
+            mbed_enabled=tuple(target_entry.get("mbed_enabled", [])),
+            build_variant=tuple(target_entry.get("build_variant", [])),
+        )
+
+
+def get_target_build_attributes(mbed_target: MbedTarget, path_to_targets_json: str) -> Any:
```

Comment:
Hmm, I'm wondering if this belongs with the other public functions rather than being here with `MbedTarget` - might be easier to see what the interface to the module is if it's all in one place.

Reply:
Define `with the other public functions`. I actually failed to group everything that takes `MbedTarget` as an argument together - planning to do it in the next step.

Reply:
Oh, I mean the `get_target_by_*` functions which are the public interface to the package.

Reply:
Hm, this has not been exposed publicly in `__init__`, so I'm not sure if this is ready to be public yet. Also, I was leaning towards having everything that takes `MbedTarget` in one place. Let's keep it where it is for the time being, unless you feel strongly otherwise. :)

Reply:
I think that's fine. We can always move it later if we need to.